### PR TITLE
Clean up ConstantPoolClassSlotIterator class

### PR DESCRIPTION
**Pull request by LinHu2016**
Use consistent data types and code formatting

### CODE REVIEW

File: runtime/gc_structs/ConstantPoolClassSlotIterator.cpp [hunk 1]
```diff
@@ -47,11 +47,11 @@ GC_ConstantPoolClassSlotIterator::nextSlot()
 			_cpDescriptionIndex = J9_CP_DESCRIPTIONS_PER_U32;
 		}
 
-		U_32 slotType = _cpDescription & J9_CP_DESCRIPTION_MASK;
+		 slotType = _cpDescription & J9_CP_DESCRIPTION_MASK;
```
Comment (dmitripivkine) [inline runtime/gc_structs/ConstantPoolClassSlotIterator.cpp:50]: missed declaration

File: runtime/gc_structs/ConstantPoolClassSlotIterator.hpp [hunk 2]
```diff
@@ -44,31 +44,30 @@ class GC_ConstantPoolClassSlotIterator
...
+	uint32_t _cpDescription;
+	uintptr_t _cpDescriptionIndex;
 	
 public:
-	GC_ConstantPoolClassSlotIterator(J9Class *clazz) :
-		_cpEntry((J9Object **)J9_CP_FROM_CLASS(clazz)),
-		_cpEntryCount(clazz->romClass->ramConstantPoolCount)
+	GC_ConstantPoolClassSlotIterator(J9Class *clazz)
+	: _cpEntry((J9Object **)J9_CP_FROM_CLASS(clazz))
+	, _cpEntryCount(clazz->romClass->ramConstantPoolCount)
```
Comment (amicic) [inline runtime/gc_structs/ConstantPoolClassSlotIterator.hpp:56]: we should preserve indentation

File: runtime/gc_structs/ConstantPoolClassSlotIterator.hpp [hunk 3]
```diff
@@ -44,31 +44,30 @@ class GC_ConstantPoolClassSlotIterator
...
 	{
 		_cpEntryTotal = _cpEntryCount;
 		if(_cpEntryCount) {
```
Comment (amicic) [inline runtime/gc_structs/ConstantPoolClassSlotIterator.hpp:59]: missing space in 'if('

File: runtime/gc_structs/ConstantPoolClassSlotIterator.hpp [hunk 4]
```diff
@@ -44,31 +44,30 @@ class GC_ConstantPoolClassSlotIterator
...
 	{
 		_cpEntryTotal = _cpEntryCount;
-		if(_cpEntryCount) {
-			_cpDescriptionSlots = SRP_PTR_GET(&clazz->romClass->cpShapeDescription, U_32 *);
+		if (_cpEntryCount) {
+			_cpDescriptionSlots = SRP_PTR_GET(&clazz->romClass->cpShapeDescription, uint32_t *);
 			_cpDescriptionIndex = 0;
 		}
-
 	};
```
Comment (amicic) [inline runtime/gc_structs/ConstantPoolClassSlotIterator.hpp:63]: I think we don't typically have semicolon after constructor

**Comment by LinHu2016** [comment]
@dmitripivkine could you please relaunch jenkins compile win jdk11? there was mistake for removing wrong semicolon. Thanks

**Comment by amicic** [comment]
jenkins compile win jdk11

File: runtime/gc_structs/ConstantPoolClassSlotIterator.hpp [hunk 5]
```diff
@@ -44,31 +44,30 @@ class GC_ConstantPoolClassSlotIterator
...
-		_cpEntry((J9Object **)J9_CP_FROM_CLASS(clazz)),
-		_cpEntryCount(clazz->romClass->ramConstantPoolCount)
+	GC_ConstantPoolClassSlotIterator(J9Class *clazz)
+		: _cpEntry((J9Object **)J9_CP_FROM_CLASS(clazz))
+		, _cpEntryCount(clazz->romClass->ramConstantPoolCount)
 	{
 		_cpEntryTotal = _cpEntryCount;
-		if(_cpEntryCount) {
-			_cpDescriptionSlots = SRP_PTR_GET(&clazz->romClass->cpShapeDescription, U_32 *);
+		if (_cpEntryCount) {
```
Comment (amicic) [inline runtime/gc_structs/ConstantPoolClassSlotIterator.hpp:59]: _cpEntryCount is an int, should we do 0 != .... ? @dmitripivkine

Comment (dmitripivkine) [reply runtime/gc_structs/ConstantPoolClassSlotIterator.hpp:59]: yes, agreed

File: runtime/gc_structs/ConstantPoolClassSlotIterator.cpp [hunk 6]
```diff
@@ -47,11 +47,11 @@ GC_ConstantPoolClassSlotIterator::nextSlot()
 			_cpDescriptionIndex = J9_CP_DESCRIPTIONS_PER_U32;
 		}
 
-		U_32 slotType = _cpDescription & J9_CP_DESCRIPTION_MASK;
+		uint32_t slotType = _cpDescription & J9_CP_DESCRIPTION_MASK;
 		J9Object **slotPtr = _cpEntry;
 
 		/* Adjust the CP slot and description information */
-		_cpEntry = (J9Object **)( ((U_8 *)_cpEntry) + sizeof(J9RAMConstantPoolItem) );
+		_cpEntry = (J9Object **)( ((uint8_t *)_cpEntry) + sizeof(J9RAMConstantPoolItem));
```
Comment (dmitripivkine) [inline runtime/gc_structs/ConstantPoolClassSlotIterator.cpp:54]: please remove space in the cast:
(J9Object **)( ((uint8_t *) -> (J9Object **)(((uint8_t *)

**Comment by amicic** [comment]
hm, seems like you lost some of the chagnes. all old INT types are back

**Comment by LinHu2016** [comment]
> hm, seems like you lost some of the chagnes. all old INT types are back

there was some rebase issue, I have updated this PR to latest, Thanks

**Comment by amicic** [comment]
jenkins compile win jdk11

**Pull request merged**